jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt

```diff
@@ -143,14 +121,19 @@ class SamProjectGeneratorSettingsPeer(val generator: SamProjectGenerator) : Proj
         settingsStep.addSettingsComponent(component)
     }
 
-    // order matters! we build the peer UI before we build the step UI,
-    // so validators should be done after BOTH have been constructed
-    fun registerValidators() {
-        // register any IDE-specific behavior
-        samInitSelectionPanel.registerValidators()
-    }
-
     override fun isBackgroundJobRunning(): Boolean = false
 
-    override fun getComponent(): JComponent = samInitSelectionPanel.mainPanel
+    // PyCharm uses this
+    override fun getComponent(locationField: TextFieldWithBrowseButton, checkValid: Runnable): JComponent = getPanel(locationField).mainPanel
+
+    // IntelliJ uses this
+    override fun getComponent(): JComponent = getPanel(null).mainPanel
+
+    private fun getPanel(locationField: TextFieldWithBrowseButton?): SamInitSelectionPanel {
+        if (!::samInitSelectionPanel.isInitialized) {
+            samInitSelectionPanel = SamInitSelectionPanel(wizardFragments, locationField)
```

Comment:
i think this is fine (as opposed to always making a new panel), but you should make a project through the wizard and then make another one in pycharm to be sure (eventually need a UI test)

Reply:
If we don't cache it never uses the selection to make the settings due to these methods may be called more than once